test/dotnet-list-p2ps.Tests/GivenDotnetListP2Ps.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using Microsoft.Build.Construction;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Msbuild.Tests.Utilities;
+using System;
+using System.IO;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.List.P2P.Tests
+{
+    public class GivenDotnetListP2Ps : TestBase
+    {
+        const string FrameworkNet451Arg = "-f net451";
+        const string ConditionFrameworkNet451 = "== 'net451'";
+        const string FrameworkNetCoreApp10Arg = "-f netcoreapp1.0";
+        const string ConditionFrameworkNetCoreApp10 = "== 'netcoreapp1.0'";
+
+        private TestSetup Setup([System.Runtime.CompilerServices.CallerMemberName] string callingMethod = nameof(Setup), string identifier = "")
+        {
+            return new TestSetup(
+                TestAssets.Get(TestSetup.TestGroup, TestSetup.ProjectName)
+                    .CreateInstance(callingMethod: callingMethod, identifier: identifier)
+                    .WithSourceFiles()
+                    .Root
+                    .FullName);
+        }
+
+        private ProjDir NewDir([System.Runtime.CompilerServices.CallerMemberName] string callingMethod = nameof(NewDir), string identifier = "")
+        {
+            return new ProjDir(TestAssetsManager.CreateTestDirectory(callingMethod: callingMethod, identifier: identifier).Path);
+        }
+
+        private ProjDir NewLib([System.Runtime.CompilerServices.CallerMemberName] string callingMethod = nameof(NewDir), string identifier = "")
+        {
+            var dir = NewDir(callingMethod: callingMethod, identifier: identifier);
+
+            try
+            {
+                new NewCommand()
+                    .WithWorkingDirectory(dir.Path)
+                    .ExecuteWithCapturedOutput("-t Lib")
+                .Should().Pass();
+            }
+            catch (System.ComponentModel.Win32Exception e)
+            {
+                throw new Exception($"Intermittent error in `dotnet new` occurred when running it in dir `{dir.Path}`\nException:\n{e}");
+            }
+
+            return dir;
+        }
+
+        private void AddFakeRef(string path, ProjDir proj)
+        {
+            new AddP2PCommand()
+                .WithProject(proj.CsProjPath)
+                .Execute($"--force \"{path}\"")
+                .Should().Pass();
+        }
+
+        [Theory]
+        [InlineData("--help")]
+        [InlineData("-h")]
+        public void WhenHelpOptionIsPassedItPrintsUsage(string helpArg)
+        {
+            var cmd = new ListP2PsCommand().Execute(helpArg);
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("Usage");
```

Comment:
true, it might actually catch some different string by accident - i'll change to "Usage: dotnet list p2ps" - I wouldn't go any further than that with validation